docs/design/014-did-doc/README.md

```diff
@@ -0,0 +1,590 @@
+# DID Doc Support in Indy
+
+## Purpose
+
+The purpose of this document is to provide vision how we can integrate w3c DIDDoc specification to Indy
+infrastructure and potentially achieve some level of interoperability with 3d party solutions that also
+follow this specification.
+
+## Use Cases
+
+As an identity owner i want to:
+
+1. create a new DID and 1 or multiple related keys
+1. publish corresonded DID Doc to Public Ledger if i have sufficient rights
+1. provide DID Doc to 3d party (Trust Ancher) for publishing to Public Ledger if i have no sufficient rights
+1. publish new DID Doc to microledger
+1. add second key to the DID DOC from a different edge device
+1. rotate a key from an edge device
+1. change authentication/authorization policies from one of the devices
+1. change service endpoints from an edge device
+1. support of multiple endpoints, each working with multiple edge devices
+1. know the recent key for this edge device to be used for signature and A2A communication using this edge device
+1. know if the key on the edge device can be used to perform modification of DID Doc on the ledger/microledger
+1. have multiple keys associated with same DID on edge device
+1. know the most recent DID DOC for the other side of communication
+1. communicate if other side of communication uses a different did method:
+    - For example, communication between entities from STN and SLN
+    - For example, Communication between Indy/Sovrin and UPort
+1. know what key(s) to use for signing/packing/A2A
+1. know what key(s) to use for verification/unpacking/A2A
+1. be able to publish existing DID Doc (maybe not as is) into a different ledger/microledger
+1. be able to re-use existing crypto material in the new approach
+1. communicate with clients using old (existing) approach
+1. use different keys for A2A and client-to-ledger communication
+
+### Create new DID and DID DOC
+
+![Alt](./create-did-ddoc.svg "Create new DID and DID DOC")
+
+### Create key on 2nd Edge Device
+
+![Alt](./create-key-2nd-device.svg "Create key on 2nd Edge Device")
+
+### Rotate a key by an Edge Device who is authenticated to do the update
+
+![Alt](./rotate-key-with-auth.svg "Rotate a key by an Edge Device who is authenticated to do the updatee")
+
+### Rotate a key by an Edge Device who is not authenticated to do the update
```

Comment:
I think you mean `not authorized` by `not authenticated`. How can a device not authorized to update still update? Do you mean updating itself? Each device should be able to update itslef without requiring any other device.

Reply:
@lovesh @dhh1128 @ashcherbakov 

1. I think you mean not authorized by not authenticated. 

Yes, it is mistake. Will fix.

2. How can a device not authorized to update still update? Do you mean updating itself? Each device should be able to update itslef without requiring any other device.

Let's consider the following use case:

1. Alice has iPhone. In wallet we she has key pair for some relationship DID that is authorized to perform DID updates (like adding new keys).
2. Alice wants to use also iPad to participate in this relationship.
3. The question is how we can achieve this? You can notice that we need to add key form new edge device (iPad) that isn't authorized to perform updates.

We suggest the following:

1. Alice generates new key pair for this DID on iPad
2. iPad application displays QR code (or doing something similar) with public key info
3. iPhone app scans QR code
4. iPhone app performs DID Doc update by adding public key from iPad
5. iPad now can be used for this relationship communications and DID Doc updates if iPhone app provided enough authorization rights for new key.

Similar use keys is creation of public DID when you need to contact Trust Anchor. It can be solved similar way.

Suggested API allows to perform described operations.

Reply:
I like the idea of keeping the pattern similar to Trust Anchor establishment at the libindy API layer. Let's leave how the verkey is transported up to consumers of the API similar to what we do for trust anchor pattern.